lib/zlib.js

```diff
@@ -130,10 +125,11 @@ function zlibBuffer(engine, buffer, callback) {
 }
 
 function zlibBufferOnData(chunk) {
-  if (!this.buffers)
+  if (!this.buffers) {
     this.buffers = [chunk];
-  else
-    ArrayPrototypePush(this.buffers, chunk);
+  } else {
+    this.buffers.push(chunk);
```

Comment:
```suggestion
    this.buffers[this.buffers.length] = chunk;
```

Reply:
This reduces readability. What is the reasoning behind this change?

Reply:
I believe it's faster, and it *also* provides the same security and reliability properties that primordials do.